test/functional/wallet_send.py

```diff
@@ -0,0 +1,339 @@
+        res = self.test_send(from_wallet=w0, to_wallet=w1, amount=51, inputs=[utxo1], add_inputs=True, add_to_wallet=False)
+        assert res["complete"]
+
+        self.log.info("Manual change address and position...")
+        self.test_send(from_wallet=w0, to_wallet=w1, amount=1, change_address="not an address",
+                       expect_error=(-5, "Change address must be a valid bitcoin address"))
+        change_address = w0.getnewaddress()
+        self.test_send(from_wallet=w0, to_wallet=w1, amount=1, add_to_wallet=False, change_address=change_address)
+        assert res["complete"]
+        res = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, add_to_wallet=False, change_address=change_address, change_position=0)
+        assert res["complete"]
+        assert_equal(self.nodes[0].decodepsbt(res["psbt"])["tx"]["vout"][0]["scriptPubKey"]["addresses"], [change_address])
+        res = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, add_to_wallet=False, change_type="legacy", change_position=0)
+        assert res["complete"]
+        change_address = self.nodes[0].decodepsbt(res["psbt"])["tx"]["vout"][0]["scriptPubKey"]["addresses"][0]
+        assert change_address[0] == "m" or change_address[0] == "n"
+
+        self.log.info("Set lock time...")
+        height = self.nodes[0].getblockchaininfo()["blocks"]
+        res = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, locktime=height + 1)
+        assert res["complete"]
+        assert res["txid"]
+        txid = res["txid"]
+        # Although the wallet finishes the transaction, it can't be added to the mempool yet:
+        hex = self.nodes[0].gettransaction(res["txid"])["hex"]
+        res = self.nodes[0].testmempoolaccept([hex])
+        assert not res[0]["allowed"]
+        assert_equal(res[0]["reject-reason"], "non-final")
+        # It shouldn't be confirmed in the next block
+        self.nodes[0].generate(1)
+        assert_equal(self.nodes[0].gettransaction(txid)["confirmations"], 0)
+        # The mempool should allow it now:
+        res = self.nodes[0].testmempoolaccept([hex])
+        assert res[0]["allowed"]
+        # Don't wait for wallet to add it to the mempool:
+        res = self.nodes[0].sendrawtransaction(hex)
+        self.nodes[0].generate(1)
+        assert_equal(self.nodes[0].gettransaction(txid)["confirmations"], 1)
+
+        self.log.info("Lock unspents...")
+        utxo1 = w0.listunspent()[0]
+        assert_greater_than(utxo1["amount"], 1)
+        res = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, inputs=[utxo1], add_to_wallet=False, lock_unspents=True)
+        assert res["complete"]
+        locked_coins = w0.listlockunspent()
+        assert_equal(len(locked_coins), 1)
+        # Locked coins are automatically unlocked when manually selected
+        self.test_send(from_wallet=w0, to_wallet=w1, amount=1, inputs=[utxo1],add_to_wallet=False)
+
+        self.log.info("Replaceable...")
```

Comment:
Added the check here in two other places just in case, though `test_send` should probably be refactored to check this unless it's explicitly expected to fail.